coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/ToDateOperatorTransformer.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino.transformers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.validate.SqlUserDefinedFunction;
+
+import com.linkedin.coral.com.google.common.base.Preconditions;
+import com.linkedin.coral.common.functions.FunctionReturnTypes;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+
+import static com.linkedin.coral.common.calcite.CalciteUtil.*;
+import static com.linkedin.coral.trino.rel2trino.CoralToTrinoSqlCallConverter.*;
+
+
+/**
+ *  This class implements the transformation from the operation of "to_date"
+ *  for example, "to_date('2023-01-01')" is transformed into "date(CAST('2023-01-01' AS TIMESTAMP))"
+ */
+public class ToDateOperatorTransformer extends SqlCallTransformer {
```

Comment:
in this transformation, we want to update the operand from a Sqldentifier, like '2023-01-01', to a SqlCall with CAST operator.

For that, we create a dummy operator "TIMESTAMP_OPERATOR" where we override the unparsing to introduce string CAST.

The logic might be ported from legacy, but is it now possible to construct the new operand by creating a SqlCall with CAST operator which casts the original operand to TIMESTAMP data type? 
It would look more intuitive and we can delete "TIMESTAMP_OPERATOR" if its not used anywhere else.

Reply:
I have tried to use `SqlCastFunction` to replace `TIMESTAMP_OPERATOR`, but the output is something like `CAST('2023-01-01' AS "TIMESTAMP")`, but the expected output is `CAST('2023-01-01' AS TIMESTAMP). So this function will be kept.

Reply:
is the difference in the quotes or something else as well? 
It looks like if the new unparsing can be understood by the engine